google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -1920,7 +1945,9 @@ public boolean equals(Object o) {
 
   @Override
   public String toString() {
-    return MoreObjects.toStringHelper(this).add("name", name).toString();
+    return MoreObjects.toStringHelper(this)
+        .add("name", bucketWithProject.getBucketName())
```

Comment:
I think we'll need to add a nullcheck and include the project id if non-null and non `_`